packages/core/src/Asset/util/addTokenMaps.ts

```diff
@@ -0,0 +1,61 @@
+import * as Cardano from '../../Cardano';
+import uniq from 'lodash/uniq';
+
+/**
+ * Given two TokenMaps, compute a TokenMap with the addition between the left-hand side and the right-hand side.
+ *
+ * @param lhs the left-hand side of the addition operation.
+ * @param rhs the right-hand side of the addition operation.
+ * @returns The addition between both TokenMaps.
+ */
+export const addMaps = (
+  lhs: Cardano.TokenMap | undefined,
+  rhs: Cardano.TokenMap | undefined
+): Cardano.TokenMap | undefined => {
+  if (!lhs) return rhs;
+  if (!rhs) return lhs;
+
+  const result = new Map<Cardano.AssetId, bigint>();
+  const intersection = new Array<Cardano.AssetId>();
+
+  for (const [key, value] of lhs.entries()) {
+    if (rhs.has(key)) {
+      intersection.push(key);
+      continue;
+    }
+
+    if (value !== 0n) result.set(key, value);
+  }
+
+  for (const [key, value] of rhs.entries()) {
+    if (lhs.has(key)) {
+      intersection.push(key);
+      continue;
+    }
+
+    if (value !== 0n) result.set(key, value);
+  }
+
+  // Elements present in both maps will be added together (lhs + rhs)
+  const uniqIntersection = uniq(intersection);
+
+  for (const id of uniqIntersection) {
+    const lshVal = lhs.get(id);
+    const rshVal = rhs.get(id);
+
+    if (lshVal! + rshVal! !== 0n) result.set(id, lshVal! + rshVal!);
+  }
+
+  return result;
+};
+
+/**
+ * Given a list of TokenMaps, compute a TokenMap with the addition of all TokenMaps.
+ *
+ * @param assets The list of TokenMaps to add.
+ */
+export const addTokenMaps = (assets: (Cardano.TokenMap | undefined)[]): Cardano.TokenMap | undefined => {
```

Comment:
This looks similar to https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/core/src/Asset/util/coalesceTokenMaps.ts#L5 . If the purpose is the same, then we should probably consolidate the 2

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1057/commits/85564707d99c7b9eccd034de4c0c841263fce521